services-directions-models/src/test/java/com/mapbox/api/directions/v5/models/RouteOptionsTest.java

```diff
@@ -622,6 +634,95 @@ public void cleanExcludes() {
     assertNull(routeOptions.exclude());
   }
 
+  @Test
+  public void putCustomRouteOptionsParamsToUrl() {
+    RouteOptions routeOptions = routeOptions().toBuilder()
+      .unrecognizedProperties(new HashMap<String, String>(){{
+        put("testName", "testValue");
+        put("testName2", "true");
+      }})
+      .build();
+
+    URL url = routeOptions.toUrl("test");
+    String query = url.getQuery();
+
+    assertTrue(
+      "url doesn't contain requested parameter" + query.toString(),
+      query.contains("testName=testValue")
+    );
+    assertTrue(
+      "url doesn't contain requested parameter" + query.toString(),
+      query.contains("testName2=true")
+    );
+  }
+
+  @Test
+  public void readCustomRouteOptionsParamsFromUrl() throws MalformedURLException {
+    String url = ROUTE_OPTIONS_URL + "&testString=test&testNumber=4.9&testBoolean=true";
+
+    RouteOptions routeOptions = RouteOptions.fromUrl(new URL(url));
+
+    Set<String> unrecognisedProperties = routeOptions.getUnrecognizedPropertiesNames();
```

Comment:
```suggestion
    Set<String> unrecognizedProperties = routeOptions.getUnrecognizedPropertiesNames();
```

nit :upside_down_face:

Reply:
I'm thinking about git hook which won't allow me to commit British english 😆